src/rpc/handlers/DepositAuthorized.cpp

```diff
@@ -71,26 +79,81 @@ DepositAuthorizedHandler::process(DepositAuthorizedHandler::Input input, Context
 
     Output response;
 
+    auto it = ripple::SerialIter{dstAccountLedgerObject->data(), dstAccountLedgerObject->size()};
+    auto sleDest = ripple::SLE{it, dstKeylet};
+    bool const reqAuth = ((sleDest.getFieldU32(ripple::sfFlags) & ripple::lsfDepositAuth) != 0u) &&
+        (sourceAccountID != destinationAccountID);
+    bool const credentialsPresent = input.credentials.has_value();
+
+    ripple::STArray authCreds;
+    if (credentialsPresent) {
+        // TODO: move this check into validation.hpp
+        if (input.credentials.value().size() > ripple::maxCredentialsArraySize)
+            return Error{Status{RippledError::rpcINVALID_PARAMS, "an array of CredentialID(hash256)"}};
+
+        for (auto const& elem : input.credentials.value()) {
+            if (!elem.is_string())
+                return Error{Status{RippledError::rpcINVALID_PARAMS, "an array of CredentialID(hash256)"}};
+
+            ripple::uint256 credHash;
+            if (!credHash.parseHex(boost::json::value_to<std::string>(elem)))
+                return Error{Status{RippledError::rpcINVALID_PARAMS, "an array of CredentialID(hash256)"}};
+
+            auto const credKeylet = ripple::keylet::credential(credHash).key;
+            auto const credLedgerObject = sharedPtrBackend_->fetchLedgerObject(credKeylet, lgrInfo.seq, ctx.yield);
+            auto credIt = ripple::SerialIter{credLedgerObject->data(), credLedgerObject->size()};
+            auto sleCred = ripple::SLE{credIt, credKeylet};
+
+            if (!credLedgerObject || (sleCred.getType() != ripple::ltCREDENTIAL) ||
+                ((sleCred.getFieldU32(ripple::sfFlags) & ripple::lsfAccepted) == 0u))
+                return Error{Status{RippledError::rpcBAD_CREDENTIALS}};
+
+            if (checkExpired(sleCred, lgrInfo.closeTime))
```

Comment:
In rippled's PR, it is `parentCloseTime`. Please double check, make sure we are not comparing with wrong time.